CONTRIBUTING.md

```diff
@@ -1,16 +1,239 @@
+  "Allow edits from maintainers" check box ticked. This will allow maintainer
+  finalizing your PR by pushing in your branch. In general, this speeds up the
+  PR merge in the main repository. Note that this is not an obligation.
+
+* Check if your code follows the
+  [coding conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions).
+  For example, you can [uncrustify](http://uncrustify.sourceforge.net) a file:
+
+      $ uncrustify -c $RIOTBASE/uncrustify-riot.cfg <your file>
+
+* RIOT provides static test tools to verify the quality of changes (flake8,
+  trailing whitespaces, documentation, etc). These tools are wrapped in a
+  single `make` target: `static-test`
+
+      $ make static-test
+
+  Use it before opening a PR to perform last time checks.
+  *watch out:* this command will rebase your branch on your master branch.
+
+* Each commit should target changes of specific parts/modules of RIOT. The
+  commits use the following pattern:
+  `path/to/module/directory: description of changes`.
+  You can use multi-line commit messages if you want to detail more the
+  changes.
+
+* Try to answer reviews as quickly as possible to speed up the review process
+  and avoid stalled PRs.
+
+You can find more information about RIOT development procedure on this
+[wiki page](https://github.com/RIOT-OS/RIOT/wiki/Development-procedures).
+
+### Git usage
+[git-usage]: #git-usage
+
+Using git is a bit difficult for newcomers. In this section, we give the bare
+minimum for a better experience with our development workflow on GitHub.
+
+#### Setup your local RIOT repository
+
+Before you start modifying code, you need to fork the RIOT upstream repository
+from the [RIOT main github page](https://github.com/RIOT-OS/RIOT).
+
+If it's your first time with git, configure your name and emails:
+
+    $ git config --global user.name = "<your name here>"
+    $ git config --global user.email = "<your email address here>"
+
+Then clone locally your fork of RIOT (replace `account name` with your actual
+login on GitHub):
+
+    $ git clone git@github.com:<account name>/RIOT.git
+
+You can keep any branch of your local repository up-to-date with the upstream
+master branch with the following commands:
+
+    $ git checkout <branch name>
+    $ git pull --rebase https://github.com/RIOT-OS/RIOT.git
+
+Use it at least before opening a PR. This will ensure the PR can be merged.
+
+#### Work on branches
+
+Avoid opening PR from the `master` branch of your fork to the master branch of
+the RIOT upstream repository: update your master branch and start a new branch
+from it.
+
+    $ git checkout master
+    $ git pull --rebase https://github.com/RIOT-OS/RIOT.git
+    $ git checkout -b <new branch>
+    # Do your changes, commit, update with latest upstream master
+    $ git push
+
+This is important because it will avoid the maintainer to have to deal with
+his master branch yours if he wants to test your changes locally.
+
+#### Add fixup commits
```

Comment:
"Add fixup commits during review" ? Just to make it more obvious when this section should be attended to.